website/source/docs/connect/discovery-chain.html.md

```diff
@@ -0,0 +1,127 @@
+---
+layout: "docs"
+page_title: "Connect - Discovery Chain (beta)"
+sidebar_current: "docs-connect-discovery_chain"
+description: |-
+  A discovery chain is a series of distinct stages that a service discovery request flows through to discover and configure Connect proxy upstreams of the service type.
+---
+
+# Discovery Chain <sup>(beta)</sup>
+
+-> **Note:** This feature is not compatible with the 
+[built-in proxy](/docs/connect/proxies/built-in.html)
+or [native proxies](/docs/connect/native.html).
+
+A discovery chain is a series of distinct stages that a service discovery
+request flows through to discover and configure Connect proxy upstreams of the
+`service` type.
+
+![diagram of discovery chain stages](/assets/images/discovery-chain-simple.svg)
+
+Each stage of this discovery chain can be dynamically reconfigured via various
+[configuration entries](/docs/agent/config_entries.html). When a configuration
+entry is missing, that stage will fall back on reasonable default behavior.
+
+## Terminology
+
+### Service Subsets
+
+There are many ways you may wish to carve up a single datacenter's pool of
+services beyond simply returning all healthy instances for load balancing.
+Canary testing, A/B tests, blue/green deploys, and soft multi-tenancy
+(prod/qa/staging sharing compute resources) all require some mechanism of
+carving out portions of the Consul catalog smaller than the level of a single
+service.
```

Comment:
The page might flow well if after this paragraph you talk very briefly about service subsets here and then describe the discovery chain as a way to route services to specific subsets. Then you can go into the services of the discovery chain stages and link people out to the config entry kinds where they can get more info... Hope that makes sense...

Reply:
So it would be:

1. Use cases
2. Subsets
3. Discovery chain allows you to discover subsets
4. Steps in the discovery chain with links to config entries

Reply:
Somewhere at the end of this paragraph, I think it would also be good to mention that you need Connect to do this stuff and why. It seems obvious because the docs are in the connect docs, but some people still could wonder.